io_u.c

```diff
@@ -1888,8 +1888,8 @@ static void account_io_completion(struct thread_data *td, struct io_u *io_u,
 				icd->error = ops->io_u_lat(td, tnsec);
 		}
 
-		if (td->o.max_latency && tnsec > td->o.max_latency)
-			lat_fatal(td, icd, tnsec, td->o.max_latency);
+		if (ddir_rw(idx) && td->o.max_latency[idx] && tnsec > td->o.max_latency[idx])
```

Comment:
This represents a change in behavior for --max_latency. Unlike the rate params which only check read, write and trim; currently the max_latency and latency_target checks currently also apply to syncs. 
This line will cause that behavior to change in the case where only one option is specified without commas - previously that would have applied the latency limit to sync also, but with this line will cause it to apply only to read, write and trim. Wanted to check what the best options are over here - 
1) Keep this change for max_latency, also introduce the ddir_rw check for latency_target so that latency checks are also consistent with BW checks, and what is logged in lat_log (syncs are not logged)
2) Keep the original functionality of max_latency for the default case - this will involve adding some way to distinguish between the case where user passes the same limits for read,write,trim vs when user passes only one value and this gets applied to other two operations in parse.c

Reply:
I'd suggest 2. Which value will sync's take for max_latency when two or more are present?

Reply:
Sure, I'll add the ddir_rw checks for latency_target too. 

This change will take take sync's out of the equation for max_latency and latency target, which is the current behavior with rate, rate_min, rate_iops etc. 

If only one option is specificifed it will apply the same value to read, write and trim.
 Two options (with either them possibly left blank using comma - eg max_latency=,10ms or max_latency=10ms,) will apply it to read and write only. Three options will apply the values to read, write and trim in that order. 

This allows us to specify trim as the fourth option (plus include in one option) in the same manner for latency and rate params if we choose to do so in the future.

Reply:
"This allows us to specify trim" do you mean "This allows us to specify sync"?

Reply:
That's correct, sorry about the typo. Corrected the above comment to say sync instead of trim.